packages/devtools_app/lib/src/service/service_manager.dart

```diff
@@ -176,6 +194,9 @@ class ServiceConnectionManager {
       _serviceAvailable = Completer();
     }
 
+    _cachedRootLibKey = null;
+    _cachedRootLibValue = null;
```

Comment:
we should null these out in vmServiceClosed, not vmServiceOpened. We can assert that they are null here if we want to be extra careful